Date: Fri, 31 Dec 2021 00:26:45 +0000
Subject: memory: fix offline_and_remove_memory use

Since commit e1c158e495661 ("mm/memory_hotplug: remove nid parameter
from remove_memory() and friends"), offline_and_remove_memory() no
longer takes a node id arguments. Adapt.

Signed-off-by: Frank van der Linden <fllinden@amazon.com>

diff --git a/drivers/base/memory.c b/drivers/base/memory.c
--- a/drivers/base/memory.c
+++ b/drivers/base/memory.c
@@ -542,7 +542,7 @@ static ssize_t remove_store(struct device *dev, struct device_attribute *attr,
 			    const char *buf, size_t count)
 {
 	u64 phys_addr;
-	int nid, ret;
+	int ret;
 	unsigned long pages_per_block = PAGES_PER_SECTION * sections_per_block;
 
 	ret = kstrtoull(buf, 0, &phys_addr);
@@ -552,8 +552,7 @@ static ssize_t remove_store(struct device *dev, struct device_attribute *attr,
 	if (phys_addr & ((pages_per_block << PAGE_SHIFT) - 1))
 		return -EINVAL;
 
-	nid = memory_add_physaddr_to_nid(phys_addr);
-	ret = offline_and_remove_memory(nid, phys_addr, MIN_MEMORY_BLOCK_SIZE * sections_per_block);
+	ret = offline_and_remove_memory(phys_addr, MIN_MEMORY_BLOCK_SIZE * sections_per_block);
 
 	if (ret)
 		return ret;
